Modified StraightDrive.java

iamef opened the pull request:
- Constructor now only takes positive powers (and the inches can be negative)
- Now, setDistance(double inches) actually uses its parameter
- Made absoluteTickGoal a int instead of a double
- Added some comments

AG6GR reviewed (approved):
Looks good to me. Check the style on line 55, space after the if and put the "throw new ..." on a new line.

AG6GR reviewed (changes requested):
Double check that commit message.

iamef committed:
Ensure turn in right direction in PointTurn.java

Moved setPower(power) to after isClockwise initialization so the robot turns in the proper direction
And after power may have set to stall power if angle < 45 degrees
